imgui_widgets.cpp

```diff
@@ -4197,7 +4197,12 @@ bool ImGui::ColorEdit4(const char* label, float col[4], ImGuiColorEditFlags flag
     if ((flags & ImGuiColorEditFlags_InputHSV) && (flags & ImGuiColorEditFlags_DisplayRGB))
         ColorConvertHSVtoRGB(f[0], f[1], f[2], f[0], f[1], f[2]);
     else if ((flags & ImGuiColorEditFlags_InputRGB) && (flags & ImGuiColorEditFlags_DisplayHSV))
+    {
+        bool restore_hue = memcmp(g.ColorPickerLastActiveColor, f, sizeof(float) * 3) == 0;
         ColorConvertRGBtoHSV(f[0], f[1], f[2], f[0], f[1], f[2]);
+        if (restore_hue)
+            f[0] = g.ColorPickerLastHue;
+    }
```

Comment:
Haven't looked/tested in detail but I wonder if it should be better to also test if saturation if <= 0 here?
Also worth sticking some comments in the restoring code to explain why we are doing all of this.

Reply:
What would i do with testing saturation? Im (mostly) clueless how these color spaces work

Reply:
When encoding an HSV value where Sat=0 into RGB, the result will be greyscale (R=G=B) and we cannot retrieve the Hue back from the RGB value.
So this "Hue restoring" code is only useful when Sat=0 so the HSV editor doesn't appear to be lossy.